snuba/query/processors/physical/hexint_column_processor.py

```diff
@@ -15,14 +15,17 @@
 
 
 class HexIntColumnProcessor(BaseTypeConverter):
-    def __init__(self, columns: Set[str]) -> None:
+    def __init__(self, columns: Set[str], size: int = 16) -> None:
         super().__init__(columns, optimize_ordering=True)
-        self._size = 16
+        self._size = size
 
     def _translate_literal(self, exp: Literal) -> Literal:
         try:
             assert isinstance(exp.value, str)
-            return Literal(alias=exp.alias, value=int(exp.value, self._size))
+            # 128 bit integers in clickhouse need to be referenced as strings
+            if self._size == 32:
+                return Literal(alias=exp.alias, value=str(int(exp.value, 16)))
+            return Literal(alias=exp.alias, value=int(exp.value, 16))
         except (AssertionError, ValueError):
             raise ColumnTypeError("Invalid hexint", should_report=False)
```

Comment:
Should this be:
```
try:
    parsed_value = int(exp.value, 16)
    return Literal(alias=exp.alias, value=str(parsed_value) if self._size == 32 else parsed_value)
except ValueError:
    raise ColumnTypeError("Invalid hexint", should_report=False)
```
Currently, the literal can be sometimes a `str` or an `int`. Does it make sense to standardize it to one type?

Reply:
Sorry, the code is wrong. I meant should it be converted to either a `str` or an `int`?